### PR TITLE
Add S3 profiles & controller mgr cfg to RamenConfig 

### PR DESCRIPTION
**Pull request by vdeenadh**
Use of the S3 profile specified in RamenConfig by DRPC and VRG will be a follow on PR.

### CODE REVIEW

File: api/v1alpha1/ramenconfig_types.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+// RamenConfig is the Schema for the ramenconfig API
+type RamenConfig struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// ControllerManagerConfigurationSpec returns the contfigurations for controllers
+	cfg.ControllerManagerConfigurationSpec `json:",inline"`
+
+	// Map of S3 store profiles
+	S3StoreProfileMap map[string]S3StoreProfile `json:"s3StoreProfileMap,omitempty"`
```
Comment (BenamarMk) [inline api/v1alpha1/ramenconfig_types.go:55]: (nit) should the name be S3StoreProfiles?

**Review by BenamarMk — APPROVED** [review]
A minor comment

**Comment by ShyamsundarR** [comment]
Also, I am not sure how we are to load the config file on demand, i.e which function/inerface to use for the same.

File: config/samples/kustomization.yaml [hunk 2]
```diff
@@ -3,4 +3,5 @@ resources:
 - ramendr_v1alpha1_drpolicy.yaml
 - ramendr_v1alpha1_drplacementcontrol.yaml
 - ramendr_v1alpha1_volumereplicationgroup.yaml
+- ramendr_v1alpha1_ramenconfig.yaml
```
[the remark this answers is not in the document]
Comment (vdeenadh) [reply config/samples/kustomization.yaml:6]: Added it.  Nevertheless, given that RamenConfig is not a CRD, I'm worry if samples can be confusing; we should think of an alternative. For now, it is in the samples folder.

File: config/manager/ramen_manager_config.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  healthProbeBindAddress: :8081
+metrics:
+  bindAddress: 127.0.0.1:8080
+webhook:
+  port: 9443
+leaderElection:
+  leaderElect: false
+  resourceName: leaderelection.ramendr.openshift.io
+  resourceNamespace: ramen-system
+s3StoreProfileMap:
```
[the remark this answers is not in the document]
Comment (vdeenadh) [reply config/manager/ramen_manager_config.yaml:13]: I created a samples file and commented out the s3StoreProfiles field in this file. But I believe the leaderElection related fields can stay. Please let me know if you disagree.

File: config/crd/kustomization.yaml [hunk 4]
```diff
@@ -5,6 +5,7 @@ resources:
 - bases/ramendr.openshift.io_volumereplicationgroups.yaml
 - bases/ramendr.openshift.io_drpolicies.yaml
 - bases/ramendr.openshift.io_drplacementcontrols.yaml
+- bases/ramendr.openshift.io_ramenconfigs.yaml
```
[the remark this answers is not in the document]
Comment (vdeenadh) [reply config/crd/kustomization.yaml:8]: Good point. Done.

File: api/v1alpha1/ramenconfig_types.go (hunk 1 above)
Comment (vdeenadh) [reply api/v1alpha1/ramenconfig_types.go:55]: Good point.  In addition, I changed the map to an array to conform with K8s API recommendation.

File: config/manager/ramen_manager_config.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+apiVersion: ramendr.openshift.io/v1alpha1
+kind: RamenConfig
+health:
+  healthProbeBindAddress: :8081
+metrics:
+  bindAddress: 127.0.0.1:8080
+webhook:
+  port: 9443
+leaderElection:
+  leaderElect: false
```
[the remark this answers is not in the document]
Comment (vdeenadh) [reply config/manager/ramen_manager_config.yaml]: Done

**Comment by vdeenadh** [comment]
> Also, I am not sure how we are to load the config file on demand, i.e which function/interface to use for the same.

I've added an interface to LoadRamenConfig().

File: PROJECT [hunk 6]
```diff
@@ -18,20 +18,27 @@ resources:
   version: v1alpha1
 - api:
     crdVersion: v1
-    namespaced: false
```
Comment (ShyamsundarR) [inline PROJECT:21]: Is this removal intentional?

File: controllers/ramenconfig.go [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+	"github.com/ghodss/yaml"
+	"github.com/go-logr/logr"
+	ramendrv1alpha1 "github.com/ramendr/ramen/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var (
+// cachedRamenConfig ramendrv1alpha1.RamenConfig
```
Comment (ShyamsundarR) [inline controllers/ramenconfig.go:30]: (nit) minor cleanups, please deal with the same in the subsequent patches on topic.

**Review by ShyamsundarR — APPROVED** [review]

File: PROJECT (hunk 6 above)
Comment (vdeenadh) [reply PROJECT:21]: I too noticed this. This change was done by operator-sdk based on its defaults I believe.

File: controllers/ramenconfig.go (hunk 7 above)
Comment (vdeenadh) [reply controllers/ramenconfig.go:30]: Yes, done.
